syntax="proto3";

package sandbox;

service Sandbox {
    rpc create_sandbox(SandboxRequest) returns (SbxCreationResponse) {}
    rpc delete_sandbox(SbxModReq) returns (Response) {}
    rpc stop_sandbox(SbxModReq) returns (Response) {}
}

message SandboxRequest {
    string name = 1;
    string tag = 2;
    string config = 3;
    repeated string images = 4;
    repeated string files = 5;
    string env_type = 6;
    string project_name = 7;
}

message SbxModReq {
    string id = 1;
}

message SbxCreationResponse {
    string message = 1;
    repeated string containers = 2;
    bool success = 3;
}

message Response {
    string message = 1;
}
